### ISSUE TITLE
Pylint failed to detect undefined variable value

### ISSUE THREAD

**Issue by socketpair** [pylint-dev/pylint#9513]
### Bug description

```python
def some_fun(value: dict[str, int]) -> int:
    try:
        first = value['first']
    except KeyError:
        pass

    return first
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
<nothing>
```

### Expected behavior

Notice that at some code paths variable value is not defined. (In our case in similar code a programmer was expected to write `return -1` instead of `pass`)

### Pylint version

```shell
pylint 3.0.4
astroid 3.0.3
Python 3.12.2 (main, Feb 21 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)]
```

### OS / Environment

Fedora 39

**Comment by jacobtylerwalls** [comment]
Duplicate of #2835. The implementation is not very hard: we just have to need to decide that it's time to create a new message for it.
